packages/@aws-cdk/aws-apigatewayv2/lib/http/stage.ts

```diff
@@ -117,4 +119,73 @@ export class HttpStage extends Resource implements IStage {
     const urlPath = this.stageName === DEFAULT_STAGE_NAME ? '' : this.stageName;
     return `https://${this.httpApi.httpApiId}.execute-api.${s.region}.${s.urlSuffix}/${urlPath}`;
   }
+
+  /**
+   * Return the given named metric for this HTTP Api Gateway Stage
+   *
+   * @default Average over 5 minutes
+   */
+  public metric(metricName: string, props?: MetricOptions): Metric {
```

Comment:
You could apply the same logic as `HttpApi` of moving this into the interface and creating an abstract `HttpStageBase` class.

This is entirely optional if you want to skip it.

Reply:
As you prefer. Currently I can import an api and get its metrics (created a new test for it) but I can't for a stage, what is correct as I cannot either associate an imported stage to an api and therefore cannot set the metric dimensions.

Reply:
It's fine. It can be done later as a separate PR.